packages/grid/x-data-grid-pro/src/hooks/features/detailPanel/useGridDetailPanelCache.ts

```diff
@@ -15,11 +20,11 @@ function cacheContentAndHeight(
   // TODO change to lazy approach using a Proxy
   // only call getDetailPanelContent when asked for an id
   const rowIds = gridRowIdsSelector(apiRef);
-  const contentCache = rowIds.reduce((acc, id) => {
+  const contentCache = rowIds.reduce((acc, id: GridRowId) => {
```

Comment:
```suggestion
  const contentCache = rowIds.reduce<Record<GridRowId, ReturnType<typeof getDetailPanelContent>>>((acc, id) => {
```

You can remove the `as` for those `reduce` that have `Record<string, something>`